CMakeLists.txt

```diff
@@ -11,7 +11,7 @@ set(OC_PKI_ENABLED ON CACHE BOOL "Enable PKI security.")
 set(OC_CLOUD_ENABLED OFF CACHE BOOL "Enable cloud communications.")
 set(OC_DEBUG_ENABLED OFF CACHE BOOL "Enable debug messages.")
 set(OC_IDD_API_ENABLED ON CACHE BOOL "Enable the Introspection Device Data API.")
-set(OC_TCP_ENABLED ON CACHE BOOL "Enable OCF communications over TCP. Necessary for Cloud communications.")
+set(OC_TCP_ENABLED OFF CACHE BOOL "Enable OCF communications over TCP. Necessary for Cloud communications.")
```

Comment:
There's a disrepancy between Makefile and CMake default configurations:

Makefile: 
TCP=0 (=OC_TCP is not defined)
OSCORE=1 (=OC_OSCORE is defined)

CMake: 
OC_TCP_ENABLED is true  (=OC_TCP is defined)
OC_OSCORE_ENABLED is false (=OC_OSCORE is not defined)

@WAvdBeek: Was this deliberate when @Oxymoron79 created the CMake or is it wrong? What should be the default setup regarding `TCP` and `OSCORE`?

Reply:
Yes, the android app is currently problematic, because it needs swig interfaces, whose generation is written in Makefile but not in CMake. But I've started looking into it last Friday.

Hmm, not sure if running all the permutations is plausible - we have at least 16 configuration switches, so that's 2^16 configurations already. :D 

What about the default values of `OC_TCP_ENABLED` and `OC_OSCORE_ENABLED` though? If the default values are not specified anywhere then I'd change the default values in CMake to match the Makefile values (=TCP disabled, OSCORE enabled).